btk/survey.py

```diff
@@ -0,0 +1,243 @@
+        Filter("i", 0.297, 20.5, 5520, 0.67, 32.36, 0.07),
+        Filter("r", 0.285, 21.2, 5520, 0.70, 43.70, 0.10),
+        Filter("g", 0.276, 22.3, 2400, 0.73, 50.70, 0.163),
+        Filter("u", 0.267, 22.9, 1680, 0.77, 9.16, 0.451),
+    ],
+)
+DES = Survey(
+    "DES",
+    pix_des,
+    [
+        Filter("i", None, None, None, None),
+        Filter("r", None, None, None, None),
+        Filter("g", None, None, None, None),
+        Filter("z", None, None, None, None),
+    ],  # Will not work properly until we fill in the correct values
+)
+CFHT = Survey(
+    "CFHT",
+    pix_cfht,
+    [
+        Filter("i", None, None, None, None),
+        Filter("r", None, None, None, None),
+    ],  # Will not work properly until we fill in the correct values
+)
+
+
+def get_filter(survey, band):
+    bands = [filt.name for filt in survey.filters]
+    assert band in bands
+    return survey.filters[bands.index(band)]
+
+
+def get_flux(ab_magnitude, filt, survey):
+    """Convert source magnitude to flux.
+    The calculation includes the effects of atmospheric extinction.
+    Credit: WeakLensingDeblending (https://github.com/LSSTDESC/WeakLensingDeblending)
+
+    Args:
+        ab_magnitude(float): AB magnitude of source.
+    Returns:
+        float: Flux in detected electrons.
+    """
+    zeropoint_airmass = 1.0
+    if survey.name == "DES":
+        zeropoint_airmass = 1.3
+    if survey.name == "LSST" or survey.name == "HSC":
+        zeropoint_airmass = 1.2
+    if survey.name == "Euclid":
+        zeropoint_airmass = 1.0
+
+    mag = ab_magnitude + filt.extinction * (filt.airmass - zeropoint_airmass)
+    return filt.exposure_time * filt.zero_point * 10 ** (-0.4 * (mag - 24))
+
+
+def get_mean_sky_level(survey, filt):
+    return get_flux(filt.sky_brightness, filt, survey) * survey.pixel_scale ** 2
+
+
+def make_wcs(pixel_scale, shape, center_pix=None, center_sky=None, projection="TAN"):
+    """Creates WCS for an image.
+    Args:
+        pixel_scale (float): pixel size in arcseconds
+        shape (tuple): shape of the image in pixels.
+        center_pix: tuple representing the center of the image in pixels
+        center_sky: tuple representing the center of the image in sky coordinates
+                     (RA,DEC) in arcseconds.
+        projection(str): projection type, default to TAN. A list of available
+                            types can be found in astropy.wcs documentation
+    Returns:
+        wcs: WCS
+    """
+    if center_pix is None:
+        center_pix = [(s + 1) / 2 for s in shape]
+    if center_sky is None:
+        center_sky = [0 for _ in range(2)]
+    w = WCS.WCS(naxis=2)
+    w.wcs.ctype = ["RA---" + projection, "DEC--" + projection]
+    w.wcs.crpix = center_pix
+    w.wcs.cdelt = [pixel_scale / 3600 for _ in range(2)]
+    w.wcs.crval = [c / 3600 for c in center_sky]
+    w.array_shape = shape
+    return w
+
+
+def get_moffat_psf(survey, filt, psf_stamp_size):
```

Comment:
I believe we need to think more about where the psf information go. This does not seem very satisfactory right now. Either we choose to use the same kind of psf for COSMOS and Catsim and we have only one `get_psf` function, or we have different ones and put `get_psf` as a method of `DrawBlendsGenerator`.

Reply:
thanks, I blended them in one function `get_psf`.